### DISCUSSION TITLE
ML.Net in MAUI

### DISCUSSION THREAD

**Discussion by Lorymi** [dotnet/maui#4153] [Q&A]
Can we use the ml.net in maui?

**Comment by summer-004** [comment]
Any updates?

**Comment by Yannikk1996** [comment]
any news?

**Comment by MhAllan** [comment]
Theoretically, you should be able to do that. Practically, here they try to make button borders without bugs.

**Reply by Yannikk1996** [reply]
I was able to get it running on windows and android. iOS works only in debug mode. In release mode it crashes with 
`System.Object Arg_TargetInvocationException InnerException: Message: PlatformNotSupported_ReflectionEmit ` on iOS

**Comment by Lemon73-Computing** [comment]
Yes, I did before.

(Documentation for ml.net in maui (this is not mine/Japanese): https://qiita.com/MAUIHighSchool/items/38e332846bec6d0bea62 )

**Comment by datis99** [comment]
Hello good time, I wrote an ebook about using Ml.net with net maui, it is complete and step by step. I published it on my LinkedIn page, here is link:
https://www.linkedin.com/posts/datis-almaspoor-a04376282_mlnet-in-net-maui-activity-7191404488635641856-iXfV?utm_source=share&utm_medium=member_android

(If the link doesn't work for any reason, search for Datis Almaspoor on LinkedIn.)
**I uploaded the file next to this message you can useI uploaded the file next to this message you can use 🚀**
[MLinMAUI.pdf](https://github.com/dotnet/maui/files/15220485/MLinMAUI.pdf)

**Comment by moljac** [comment]
I used it with Xamarin.Forms, so it must be doable.

ML.net is `netstandard2.0` package

https://www.nuget.org/packages/Microsoft.ML

During learning and testing model[s] move ML code to library and test it from console app and then reuse library from MAUI app. This is how I did it few years back.
